src/libexpr/eval.cc

```diff
@@ -957,6 +957,15 @@ void Value::mkString(std::string_view s, const NixStringContext & context)
     copyContextToValue(*this, context);
 }
 
+void Value::mkEmptyString(const Value & other)
+{
+    mkString("");
+
+    NixStringContext context;
+    copyContext(other, context);
+    copyContextToValue(*this, context);
```

Comment:
We have two representations for string context: the well typed `NixStringContext` and the garbage collected C string representation. For the latter it's sufficient to just assign the pointer, without copying the referenced value.
ie:

```suggestion
    assert(internalType == tString);
    string.context = other.string.context;
```

Reply:
When I was tracking down the issue I had last night, I got paranoid that the contexts where getting mutated somewhere I didn't understand, but it turned out to just be that I can't use `<=` for len. Reverted.